### PR TITLE
[build] Select uint63 using `ocamlc -config` variables.

### PR DESCRIPTION
**Pull request by ejgallego**
This seems more robust and avoids having another implementation of
`cp`.

### CODE REVIEW

File: kernel/dune [hunk 1]
```diff
@@ -14,15 +14,10 @@
...
-(executable
-  (name write_uint63)
-  (modules write_uint63)
-  (libraries unix))
-
 (rule
  (targets uint63.ml)
- (deps (:gen ./write_uint63.exe) uint63_x86.ml uint63_amd64.ml)
- (action (run %{gen})))
+ (deps (:gen-file uint63_%{ocaml-config:architecture}_%{ocaml-config:int_size}.ml))
```
Comment (SkySkimmer) [inline kernel/dune]: While we're at it, why do we depend on `architecture`? Those files don't have any assembly AFAICT

File: kernel/dune [hunk 2]
```diff
@@ -14,15 +14,10 @@
...
+ (action (copy %{gen-file} %{targets})))
```
Comment (SkySkimmer) [inline kernel/dune]: Why not `copy#`?

File: kernel/write_uint63.ml [hunk 3]
```diff
@@ -31,8 +31,8 @@ let ml_file_copy input output =
 
 let write_uint63 () =
   ml_file_copy
-    (if max_int = 1073741823 (* 32-bits *) then "uint63_x86.ml"
-     else (* 64 bits *) "uint63_amd64.ml")
+    (if max_int = 1073741823 (* 32-bits *) then "uint63_i386_31.ml"
+     else (* 64 bits *) "uint63_amd64_63.ml")
     "uint63.ml"
 
 let () = write_uint63 ()
```
Comment (SkySkimmer) [inline kernel/write_uint63.ml]: I guess this is kept for make
Alternatively we could dispatch in configure, ie
- configure writes `INT_SIZE` to config/Makefile
- Makefile.build copies `uint63_$(INT_SIZE)`

**Comment by SkySkimmer** [comment]
@ejgallego can you respond to my earlier comments on this PR?

File: kernel/dune (hunk 1 above)
Comment (ejgallego) [reply kernel/dune]: No idea, I just followed previous naming.

Comment (ejgallego) [reply kernel/dune]: cc @vbgl

File: kernel/write_uint63.ml (hunk 3 above)
Comment (ejgallego) [reply kernel/write_uint63.ml]: Indeed, this is kept for make, as well as the copy program.

**Comment by ejgallego** [comment]
> @ejgallego can you respond to my earlier comments on this PR?

Done, sorry.

File: kernel/dune (hunk 1 above)
Comment (vbgl) [reply kernel/dune]: Indeed, this information seems redundant. What we really care about here is the int size.

**Review by gares — APPROVED** [review]

**Review by Zimmi48 — COMMENTED** [review]
Can we get an issue to remember to remove the copy program and the `write_uint63.ml` file when the Makefiles are removed? And can you remove the architecture name from the files since it is apparently redundant?

**Review by maximedenes — APPROVED** [review]

**Comment by SkySkimmer** [comment]
This needs an assignee

**Review by vbgl — APPROVED** [review]

**Comment by maximedenes** [comment]
I had missed the file name changes. What was the motivation for adding a suffix?

**Pull request merged**